components/ui/location-combobox.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from "react";
+import { Check, ChevronsUpDown } from "lucide-react";
+
+import { cn } from "@/lib/utils";
+import { Button } from "@/components/ui/button";
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+} from "@/components/ui/command";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+
+type Location = {
+  name: string;
+  type: string;
+  regionId: string;
+  coordinates: [number, number];
+  country: string;
+  domain: string;
+};
+
```

Comment:
```suggestion
import { APIRegionArrayFormatted } from "@/app/api/hotels/region/route";
```

`APIRegionArrayFormatted`. This is the region API's structure. Use this exact structure when creating cacheLocations. Or retrieve them from the API directly through Postman.